packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -448,7 +488,102 @@ export const prepareCosmosOrchestrationAccountKit = (
             Fail`'amountToCoin' not working for ${q(amount.brand)} until #10449; use 'DenomAmount' for now`;
           return coerceCoin(chainHub, amount);
         },
+        /**
+         * Submit a transaction on behalf of the remote account for execution on
+         * the remote chain.
+         *
+         * Set `relativeTimeoutNs` to provide a timeout for the IBC packet.
+         *
+         * `TxBody` fields like `timeoutHeight` and `memo` can be set, but these
+         * typically do not affect IBC app protocols like PFM, ICA.
+         *
+         * @param {[
+         *   ...Parameters<HostOf<IcaAccount['executeEncodedTx']>>,
+         *   nextWatcher?: {
+         *     onFulfilled<T>(resultMeta: {
+         *       result: PromiseVow<string>;
+         *       meta?: Record<string, any>;
+         *     }): T;
+         *   },
+         * ]} args
+         * @returns {Vow<{ result: Vow<string>; meta: Record<string, any> }>}
+         *   base64 encoded bytes string. Can be decoded using the corresponding
+         *   `Msg*Response` object.
+         * @throws {Error} if packet fails to send or an error is returned
+         */
+        executeEncodedTxWithMeta(...args) {
+          return asVow(() => {
+            const [msgs, opts = {}, nextWatcher] = args;
+            const { chainAddress } = this.state;
+            const { helper } = this.facets;
+            const conn = helper.owned();
+            const result = watch(E(conn).executeEncodedTx(msgs, opts));
+            const all = [];
+            if (connectionToDetails.has(conn)) {
+              const { agoric, la, counterparty, ra } =
+                connectionToDetails.get(conn);
+              all.push(agoric, la, counterparty, ra);
+            } else {
+              const agoric = chainHub.getChainInfo('agoric');
+              const la = E(conn).getLocalAddress();
+              const counterparty = `cosmos:${chainAddress.chainId}`;
+              const ra = E(conn).getRemoteAddress();
+
+              all.push(agoric, la, counterparty, ra);
+            }
+            return watch(
+              allVows(all),
+              this.facets.attachTxMetaWatcher,
+              nextWatcher,
+              { result, meta: {} },
+              'ibc',
+            );
+          });
+        },
+      },
+      attachTxMetaWatcher: {
+        onFulfilled(
+          [agoric, la, counterparty, ra],
+          nextWatcher,
+          { result, meta: origMeta },
+          protocol,
+        ) {
+          const { helper } = this.facets;
+          const conn = helper.owned();
+          if (!connectionToDetails.has(conn)) {
+            // Memoize the details for next time.
+            connectionToDetails.init(
```

Comment:
Tell me about the size/growth of `connectionToDetails` please? does it ever shrink?

It's not mentioned in **Scaling Considerations**; was that because it's unremarkable? I don't yet understand it well enough for that to be obvious to me.

Reply:
oh... it's weak. hm.